### PR TITLE
add ability to add or remove rep

### PR DESCRIPTION
**Pull request by mp-singh**

### CODE REVIEW

File: bin/bot/src/discord/slash/reputation.rs [hunk 1]
```diff
@@ -25,6 +29,62 @@ pub fn register(command: &mut CreateApplicationCommand) -> &mut CreateApplicatio
...
+                        .description("The member that the event is for (use Brodsky if ambiguous)")
+                        .kind(CommandOptionType::User)
+                        .required(true)
+                })
+                .create_sub_option(|option| {
+                    option
+                        .name("amount")
+                        .description("The amount of reputation to add")
+                        .kind(CommandOptionType::Integer)
+                        .min_int_value(1)
```
Comment (mp-singh) [inline bin/bot/src/discord/slash/reputation.rs:49]: should these also have a max value?

File: bin/bot/src/discord/slash/reputation.rs [hunk 2]
```diff
@@ -85,7 +200,7 @@ async fn view(
         }
     )
     .await else {
-        return interaction.reply("Failed to transfer money").await;
+        return interaction.reply("Failed to retrieve current reputation").await;
```
Comment (mp-singh) [inline bin/bot/src/discord/slash/reputation.rs:203]: this is just cleanup

File: bin/bot/src/discord/slash/reputation.rs [hunk 3]
```diff
@@ -64,13 +124,68 @@ pub async fn run(ctx: &Context, command: &ApplicationCommandInteraction) -> sere
...
+    let mut interaction = Interaction::new(ctx, Generic::Application(command), options);
+    super::requires_roles(
+        command.user.id,
+        &[STAFF],
+        &command
+            .member
+            .as_ref()
+            .expect("member should always exist on guild commands")
+            .roles,
+        ShouldAsk::Deny,
```
Comment (mp-singh) [inline bin/bot/src/discord/slash/reputation.rs:151]: im assuming this is what creates a button to allow or deny a command.
Should this be turned to allow?

**Review by BrettMayson — APPROVED** [review]

File: bin/bot/src/discord/slash/reputation.rs (hunk 1 above)
Comment (BrettMayson) [reply bin/bot/src/discord/slash/reputation.rs:49]: I think it is fine to not

**Pull request merged**